### PR TITLE
Abe/csx recursion repeater

### PR DESCRIPTION
**Pull request by a-hend**
It seems like recursion can modify a value with each call of itself and then return the modified value one the breaking condition has been met (such as Attempt #2)

-or-

Recursion can call itself until it meets a breaking condition (usually with some kind or counter that is modified with each call) and then return a value will then be modified upon each return statement (such as Attempt #1)

Am I understanding this correctly?  
Communicating my thoughts clearly or poorly?
If I am understanding this correctly: Of the two options, is there a preference for how to use recursion?

### CODE REVIEW

**Comment by C-STYR** [comment]
To borrow a common phrase:

To understand recursion you must first understand recursion. 

Seems like you're almost there, I won't step in.  Give it time to gel. 

Regarding preference, go with whatever is simpler and easier to read.

**Comment by a-hend** [comment]
Ha, or like the google search for 'recursion' that returns: "did you mean 'recursion'?"

Am I understanding correctly that Attempt #1 modifies `char` upon returning from its breaking condition?

The way I'm reading it is something like this:
char + repeater(char)
char + repeater(char) + repeater(char)
char + repeater(char) + repeater(char) + repeater(char) 
char + repeater(char) + repeater(char) + repeater(char) + repeater(char) *Breaking Condition Met*

Then on that fifth calling of `repeater(char)` the breaking condition is met, and the parameter and argument is matched and `char` suddenly becomes `g`

So now it reads something like:
char + repeater(char) + repeater(char) + repeater(char) + 'g'
char + repeater(char) + repeater(char) + 'g' + 'g'
char + repeater(char) + 'g' + 'gg'
char + 'g'+ 'ggg'
'g' + 'gggg'
'ggggg'

In other words the value of `char` is returned and modified only after the the recursive function has been called until it met its breaking condition.

Am I understanding this example correctly?

File: Users/abehenderson/CSX/Recursion/repeater.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// Attempt #2 (CSX Solution)
+
+function repeater(char) {
```
Comment (C-STYR) [inline Users/abehenderson/CSX/Recursion/repeater.js]: This line contains the function signature for the parameter `char`: it is whatever is passed in.

File: Users/abehenderson/CSX/Recursion/repeater.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  if (char.length === 5) return char;
+    return repeater(char + char[0]);
```
Comment (C-STYR) [inline Users/abehenderson/CSX/Recursion/repeater.js]: When `repeater` is called here, the argument `char`'s value is, in the first run, `gg`, in the second run `ggg` ....in the fourth run `ggggg` where it is returned (backwards down the call stack).

**Review by C-STYR — APPROVED** [review]

File: Users/abehenderson/CSX/Recursion/repeater.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+// let counter = 0;
+
+// function repeater(char) {
+//   counter++;  
+//   if (counter >= 5) {
+//     counter = 0;
+//     return char;
+//   } else {
+//     return char + repeater(char);
```
Comment (C-STYR) [inline Users/abehenderson/CSX/Recursion/repeater.js]: On the first call, the return value is `g + repeater(g)`... does this attempt even work? I'm not sure just looking at it. I can see a place it might break down.

**Review by C-STYR — APPROVED** [review]

File: Users/abehenderson/CSX/Recursion/repeater.js (hunk 3 above)
Comment (a-hend) [reply Users/abehenderson/CSX/Recursion/repeater.js]: Yes it worked for me.  But, are you saying that `return repeater(char + char[0];` is a better syntax to use?

File: Users/abehenderson/CSX/Recursion/repeater.js (hunk 2 above)
Comment (a-hend) [reply Users/abehenderson/CSX/Recursion/repeater.js]: Ok so with every stack frame push, `char` is modified, and the final result `ggggg`is returned (because the breaking condition has been met) back to each previous stack frame down the call stack until it is returned to global memory?

File: Users/abehenderson/CSX/Recursion/repeater.js (hunk 1 above)
Comment (a-hend) [reply Users/abehenderson/CSX/Recursion/repeater.js]: Oh, right, because the first thing that happens upon invoking a function is matching parameters with arguments.

File: Users/abehenderson/CSX/Recursion/repeater.js (hunk 2 above)
Comment (C-STYR) [reply Users/abehenderson/CSX/Recursion/repeater.js]: yes
